### PR TITLE
refactor

### PR DESCRIPTION
**Pull request by mangkoran**
WIP. I decided to do some refactor of the codebase so it looks more neat (IMO). Apparently it will solve https://github.com/jskherman/cv.typ/pull/1#issuecomment-1910896918 #9, #10 but not yet for now.

Current state/commit is on parity with master (no failed compilation).

Please let me know if you have any suggestions.

### CODE REVIEW

**Comment by mangkoran** [comment]
Alright I think it's now ready for review. No issue so far on my end. Friendly mention @jskherman @kyawzazaw @phamson02

**Comment by mangkoran** [comment]
I also have an idea to split input files/YAML for sensitive data such as phone number, email, location etc. So we could track the other input file while keeping the sensitive data locally. What do you think about it?

File: cv.typ [hunk 1]
```diff
@@ -2,341 +2,391 @@
...
+          let end = utils.str_to_date(p.endDate)
+
+          if index != 0 { v(0.5em) }
+
+          block(above: 0.5em, {
+            line_two(p.position, right_start: start, right_end: end)
+          })
+
+          for hi in p.highlights {
+            [- #hi]
```
Comment (jskherman) [inline cv.typ:161]: Hm, this part should be reverted back to using `eval()`. This breaks the use of markup like bold or italics when present in the YAML file.

```suggestion
        for hi in p.highlights{
            [- #eval(hi, mode: "markup")]
        }
```

File: cv.typ [hunk 2]
```diff
@@ -2,341 +2,391 @@
...
+          line_one(org.organization, right: org.location, url: org.url)
+
+          linebreak()
+          // line 2: position and date
+          line_two(org.position, right_start: start, right_end: end)
+
+          if org.at("highlights", default: none) != none {
+            for h in org.highlights {
+              [- #h]
+            }
```
Comment (jskherman) [inline cv.typ:233]: Similar problem here as well as above...

```suggestion
            for hi in org.highlights {
              [- #eval(hi, mode: "markup")]
            }
```

File: cv.typ [hunk 3]
```diff
@@ -2,341 +2,391 @@
...
+          // line 1: project name
+          line_one(project.name, url: url)
+
+          linebreak()
+          // line 2: organization and date
+          line_two(project.affiliation, right_start: start, right_end: end)
+
+          for h in project.highlights {
+            [- #h]
+          }
```
Comment (jskherman) [inline cv.typ:264]: `eval()` 3:

```suggestion
            for hi in project.highlights {
              [- #eval(hi, mode: "markup")]
            }
```

File: cv.typ [hunk 4]
```diff
@@ -2,341 +2,391 @@
...
+          line_one(award.title, right: award.location, url: url)
+
+          linebreak()
+          // line 2: issuer and date
+          line_two(award.issuer, right_end: date)
+
+          if award.at("highlights", default: none) != none {
+            for h in award.highlights {
+              [- #h]
+            }
```
Comment (jskherman) [inline cv.typ:290]:
```suggestion
            for hi in award.highlights {
              [- #eval(hi, mode: "markup")]
            }
```

File: cv.typ [hunk 5]
```diff
@@ -2,341 +2,391 @@
...
+      for lang in langs {
+        langs_join.push([#lang.language (#lang.fluency)])
+      }
+
+      dot_list("Languages", langs_join)
+
+      let skills = info.at("skills", default: none)
+      for skill in skills {
+        dot_list(skill.category, skill.skills)
+      }
```
Comment (jskherman) [inline cv.typ:358]: To reduce confusion, might I suggest to use the variable name `group`, `skill-group`, or `skill_group`?

```suggestion
      for skill_group in skills {
        dot_list(skill_group.category, skill_group.skills)
      }
```

File: cv.typ [hunk 6]
```diff
@@ -2,341 +2,391 @@
...
-        }
-    ]} else {}
+  if info.at("references", default: none) != none {
+    block({
+      [== References]
+
+      for ref in info.references {
+        let foo = ()
+        foo.push(ref.reference)
+        dot_list(ref.name, foo, url: ref.url)
```
Comment (jskherman) [inline cv.typ:374]: May I ask if `foo` here is needed? Could it instead be an [array created directly](https://typst.app/docs/reference/foundations/array/#constructor)?

```suggestion
        dot_list(ref.name, (ref.reference,), url: ref.url)
```

**Comment by jskherman** [comment]
> I also have an idea to split input files/YAML for sensitive data such as phone number, email, location etc. So we could track the other input file while keeping the sensitive data locally. What do you think about it?

Hm, technically you could consider _all_ of the data in a résumé or CV sensitive data.  Also, it might get a bit too unwieldy to have the data scattered in different YAML files. To illustrate, my current use case involves using different YAML files slightly altered from a main YAML file (used in the CV) to create one/two-page résumés that are tailored to a specific job description or role. For me, having one YAML file per document makes it easier to diff these customized "versions".

**Comment by mangkoran** [comment]
Hi @jskherman I apologize for the late reply as I am busy with with IRL. Originally this PR should be opened as a draft PR as it still need some rework. And thank you for leaving suggestions.

However, I'm planning to do some additional refactor to comply with Typst's [template format](https://github.com/typst/templates). I think I will still use this PR for that.

**Comment by mangkoran** [comment]
Suggestions committed. Could you please recheck whether it's what you requested?

**Comment by jskherman** [comment]
Hi @mangkoran, sorry for only getting back to you at this time since I had a big exam to prepare for in the past few months... I'm sorry to spring this on you only now after you have done a lot of work on this PR. But I was thinking of closing this PR and doing a double check on the code base as a whole for consistency (including reacquainting myself with the code after being away for a while) and, while at it, convert the setup to use the new system for templates in the Typst Universe. To be honest, it also feels rather daunting to review such a huge PR since I'm only coding as a hobby (skill issue, I know). I'm quite thankful though for the amount of work you put into this.

**Pull request closed without merging**